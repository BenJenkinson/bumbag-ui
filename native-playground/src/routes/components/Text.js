import React from 'react';
import { Box, Text } from 'bumbag-native';
import { Preview } from '../../components/Preview';
import { PreviewSection } from '../../components/PreviewSection';

export default function App() {
  return (
    <Box>
      <Text fontSize="20px" fontWeight="500" marginBottom="16px">
        Text
      </Text>
      <PreviewSection title="Basic">
        <Preview>
          <Box>
            <Text use="strong">
              Hello world, this is a bit of <Text fontWeight="500">text</Text>.
            </Text>
          </Box>
        </Preview>
      </PreviewSection>
    </Box>
  );
}
