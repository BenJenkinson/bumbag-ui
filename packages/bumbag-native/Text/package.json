{
  "name": "bumbag-native/Text",
  "private": true,
  "sideEffects": false,
  "main": "../lib/Text",
  "module": "../es/Text",
  "types": "../ts/Text"
}