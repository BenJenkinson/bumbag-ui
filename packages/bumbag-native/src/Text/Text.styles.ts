import { Platform, Text } from 'react-native';
import { styled } from '../styled';
import { palette, theme } from '../utils/theme';

export const StyledText = styled(Text)`
  color: ${palette('text')};

  ${Platform.OS === 'web'
    ? `
      font-family: inherit;
      font-size: unset;
    `
    : ''}

  ${theme('native.Text', 'styles.base')};
`;
