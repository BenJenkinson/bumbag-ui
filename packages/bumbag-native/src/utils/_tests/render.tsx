import * as React from 'react';
import { ThemeConfig } from '../../types/theme';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (
  Component: any,
  { colorMode = 'default', theme = {} }: { colorMode?: string; theme?: ThemeConfig } = {}
) =>
  render(
    <Provider platform="native" theme={theme} colorMode={colorMode}>
      {Component}
    </Provider>
  );
