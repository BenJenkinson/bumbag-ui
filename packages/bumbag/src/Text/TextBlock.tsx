import { Box as ReakitBox } from 'reakit';

import { useClassName, createComponent, createElement, createHook } from '../utils';
import { Box, BoxProps } from '../Box';

import * as styles from './styles';

export type LocalTextBlockProps = {};
export type TextBlockProps = BoxProps & LocalTextBlockProps;

const useProps = createHook<TextBlockProps>(
  (props, { themeKey, themeKeyOverride }) => {
    const boxProps = Box.useProps(props);

    const className = useClassName({
      style: styles.TextBlock,
      styleProps: props,
      themeKey,
      themeKeyOverride,
      prevClassName: boxProps.className,
    });

    return { ...boxProps, className };
  },
  { themeKey: 'TextBlock' }
);

export const TextBlock = createComponent<TextBlockProps>(
  (props) => {
    const textBlockProps = useProps(props);
    return createElement({ children: props.children, component: ReakitBox, use: props.use, htmlProps: textBlockProps });
  },
  {
    attach: {
      useProps,
      displayName: 'TextBlock',
    },
    defaultProps: {
      use: 'span',
    },
    themeKey: 'TextBlock',
  }
);
