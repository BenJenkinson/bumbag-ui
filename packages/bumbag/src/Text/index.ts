import * as styles from './styles';

export * from './Text';
export { styles as textStyles };
