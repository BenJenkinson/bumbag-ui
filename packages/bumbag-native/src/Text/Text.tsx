import { TextProps as RNTextProps } from 'react-native';
import { createComponent, createElement, createHook } from 'bumbag/utils';

import { Box, BoxProps } from '../Box';
import * as styles from './Text.styles';

export type LocalTextProps = {};
export type TextProps = BoxProps & LocalTextProps;

const useProps = createHook<TextProps>(
  (props) => {
    const boxProps = Box.useProps(props);
    return { ...boxProps };
  },
  { themeKey: 'Text' }
);

export const Text = createComponent<RNTextProps & TextProps>(
  (props) => {
    const textProps = useProps(props);
    return createElement({
      children: props.children,
      component: styles.StyledText,
      htmlProps: textProps,
    });
  },
  {
    attach: {
      useProps,
      displayName: 'Text',
    },
    themeKey: 'Text',
  }
);
