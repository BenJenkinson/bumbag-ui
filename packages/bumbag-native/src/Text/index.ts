import * as textStyles from './Text.styles';

export * from './Text';
export { textStyles };
