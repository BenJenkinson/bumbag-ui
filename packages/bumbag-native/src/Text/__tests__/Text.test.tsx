import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Text } from '../Text';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Text>Hello world</Text>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Text color="primary">Hello world</Text>);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('hook', () => {
    it('should return with Text props', () => {
      const { result } = renderHook(() => Text.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(<Text>{(TextProps) => <Text {...TextProps}>Hello world</Text>}</Text>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('theming', () => {
  it('Text.styles.base should render correctly', () => {
    const { container } = render(<Text>hello world</Text>, {
      theme: { Text: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Text.styles.base should render correctly with default font', () => {
    const { container } = render(<Text>hello world</Text>, {
      theme: {
        fonts: { default: 'default', heading: 'heading' },
        Text: { styles: { base: { backgroundColor: 'red' } } },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Text.styles.base should render correctly with heading font', () => {
    const { container } = render(<Text font="heading">hello world</Text>, {
      theme: {
        fonts: { default: 'default', heading: 'heading' },
        Text: { styles: { base: { backgroundColor: 'red' } } },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for color', () => {
    const { container } = render(<Text>hello world</Text>, {
      theme: { Text: { defaultProps: { color: 'primary' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
