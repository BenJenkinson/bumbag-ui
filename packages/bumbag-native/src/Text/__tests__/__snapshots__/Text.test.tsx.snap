// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition hook should return with Text props 1`] = `
Object {
  "active": false,
  "colorMode": "default",
  "elementRef": [Function],
  "focus": false,
  "hover": false,
  "hoveractive": false,
  "ref": [Function],
  "style": Array [
    Object {
      "fontSize": 200,
    },
  ],
}
`;

exports[`composition render props should render correctly 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="font-size: 16px;"
>
  Hello world
</div>
`;

exports[`defaultProps should render correctly for color 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="color: rgb(87, 79, 235); font-size: 16px;"
>
  hello world
</div>
`;

exports[`props should render correctly 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="font-size: 16px;"
>
  Hello world
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="color: rgb(87, 79, 235); font-size: 16px;"
>
  Hello world
</div>
`;

exports[`theming Text.styles.base should render correctly 1`] = `
<div
  class="css-text-901oao"
  dir="auto"
  style="background-color: rgb(255, 0, 0); font-size: 16px;"
>
  hello world
</div>
`;
